### ISSUE TITLE
Simulating the same workflow but getting error while start training.

### ISSUE THREAD

**Issue by Ibrokhimsadikov** [walidamamou/relation_extraction_transformer#5]
Hello @walidamamou ,  

I would be pleased if you can help me to understand where I am having issue while using the same workflow as yours:
1) I have prepared my own annotations as the text files you shared on this github repo: relations_dev.txt, relations_train.txt, relations_test.txt exactly the same format as yours (spacy version I am using is the latest on pip)

2: I use binary converter python file to convert txt files into binary file. the only change, I made on Map labels in my case I have one TEST-RESULT relation

3) When I start traning then I am getting the following weird error

================================= train_cpu =================================
Running command: /home/ec2-user/anaconda3/envs/Python_38/bin/python3.8 -m spacy train configs/rel_tok2vec.cfg --output training --paths.train data/relations_training.spacy --paths.dev data/relations_dev.spacy -c ./scripts/custom_functions.py
ℹ Using CPU

=========================== Initializing pipeline ===========================
[2021-09-13 14:25:40,953] [INFO] Set up nlp object from config
[2021-09-13 14:25:40,960] [INFO] Pipeline: ['tok2vec', 'relation_extractor']
[2021-09-13 14:25:40,962] [INFO] Created vocabulary
[2021-09-13 14:25:40,962] [INFO] Finished initializing nlp object
[2021-09-13 14:25:53,608] [INFO] Initialized pipeline components: ['tok2vec', 'relation_extractor']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['tok2vec', 'relation_extractor']
ℹ Initial learn rate: 0.001
E    #       LOSS TOK2VEC  LOSS RELAT...  REL_MICRO_P  REL_MICRO_R  REL_MICRO_F  SCORE 
---  ------  ------------  -------------  -----------  -----------  -----------  ------
⚠ Aborting and saving the final best model. Encountered exception:
KeyError("[E900] Could not run the full pipeline for evaluation. If you
specified frozen components, make sure they were already initialized and
trained. Full pipeline: ['tok2vec', 'relation_extractor']")
Traceback (most recent call last):
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/training/loop.py", line 267, in evaluate
    scores = nlp.evaluate(dev_corpus(nlp))
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/language.py", line 1362, in evaluate
    results = scorer.score(examples)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/scorer.py", line 129, in score
    scores.update(component.score(examples, **self.cfg))
  File "/home/ec2-user/SageMaker/Spacyv3/radrelex/rel_component/scripts/rel_pipe.py", line 201, in score
    return score_relations(examples, self.threshold)
  File "/home/ec2-user/SageMaker/Spacyv3/radrelex/rel_component/scripts/rel_pipe.py", line 211, in score_relations
    gold_labels = [k for (k, v) in gold[key].items() if v == 1.0]
KeyError: (3, 10)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/__main__.py", line 4, in <module>
    setup_cli()
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/cli/_util.py", line 69, in setup_cli
    command(prog_name=COMMAND)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/typer/main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/cli/train.py", line 59, in train_cli
    train(nlp, output_path, use_gpu=use_gpu, stdout=sys.stdout, stderr=sys.stderr)
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/training/loop.py", line 114, in train
    raise e
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/training/loop.py", line 98, in train
    for batch, info, is_best_checkpoint in training_step_iterator:
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/training/loop.py", line 212, in train_while_improving
    score, other_scores = evaluate()
  File "/home/ec2-user/anaconda3/envs/Python_38/lib/python3.8/site-packages/spacy/training/loop.py", line 269, in evaluate
    raise KeyError(Errors.E900.format(pipeline=nlp.pipe_names)) from e
KeyError: "[E900] Could not run the full pipeline for evaluation. If you specified frozen components, make sure they were already initialized and trained. Full pipeline: ['tok2vec', 'relation_extractor']"

**Comment by Ibrokhimsadikov** [comment]
Please share your thoughts, is this training data related or not

**Comment by walidamamou** [comment]
It looks like an offset alignment issue, was the training data generated with spacy 3?

**Comment by Ibrokhimsadikov** [comment]
Yes I used Spacy version 3:

And my training data looks like that as txt file:

{"document": "there is no acute osseous abdomen body ", "tokens": [{"text": "osseous abdomen body", "start": 18, "end": 38, "token_start": 4, "token_end": 6, "entityLabel": "OBSERVATION"}, {"text": "there is no", "start": 0, "end": 11, "token_start": 0, "token_end": 2, "entityLabel": "MODIFIER"}], "relations": [{"child": 4, "head": 0, "relationLabel": "TEST_RESULT"}]}


and your txt file looks like that:

{
    "document": "3+ years Swift & Objective-C and experience with iOS internals Experience building an entire app from scratch and ideally a portfolio of apps featured in the App Store Someone who knows every trick in the book on UI transitions, network communication and memory/battery efficiency Strong UI/design skill experience is a plus",
    "tokens": [
      {
        "text": "3+ years",
        "start": 0,
        "end": 8,
        "token_start": 0,
        "token_end": 2,
        "entityLabel": "EXPERIENCE"
      },
      {
        "text": "Swift & Objective-C",
        "start": 9,
        "end": 28,
        "token_start": 3,
        "token_end": 7,
        "entityLabel": "SKILLS"
      }
    ],
    "relations": [{ "child": 3, "head": 0, "relationLabel": "EXPERIENCE_IN" }]
  },

**Comment by walidamamou** [comment]
Did you install spacy 3 when running the conversion? Do you mind sending your training file?

**Comment by Ibrokhimsadikov** [comment]
@walidamamou how can I share that file

**Comment by walidamamou** [comment]
Can you provide some sharable link, like on google drive

**Comment by Ibrokhimsadikov** [comment]
Dear @walidamamou , Unfortunately it is not possible to share the data according to NDA. May be there is any other way to understand the issue? I was originally using overall 31000 sentences dev+test+train, and I was getting this error. When I simply tried to test with 100 sentences dev+test+train I was able to train the model and everything working fine as is in your article. But when i tried to train with 1000, 10000 I am still getting the error.

I think the main error is here  ->   KeyError: (3, 10) as stated in error log, would it be kind of out of vocabulary error, or still alignment error?
